rgw: implement support for authentication using keystone for s3 and swift (#13807)
rook/rook: Documentation/Storage-Configuration/Object-Storage-RGW/object-storage.md

## Configure an Object Store

Rook has the ability to either deploy an object store in Kubernetes or to connect to an external RGW service.
Most commonly, the object store will be configured in Kubernetes by Rook.

> Alternatively, if you have an existing Ceph cluster with [Rados Gateways](https://docs.ceph.com/en/quincy/radosgw/index.html), see the [external section](#connect-to-an-external-object-store) to consume it from Rook.

Reviewer: Please adjust based on "you" usage here.

Author: fa106d4e19f711f3493b7d8f34791f46dc51e4cc

Reviewer: Also, as a preparation, we will ask you to squash all the commits before final merge. It's fine to keep each commit separate while we are working, however.

Author: Seems as if this commit has already been squased already. Trying to cherry-pick it I ended up with an empty commit.
If you could please do a quick check whether the "you" based language is not gone.

Rook can configure the Ceph Object Store for several different scenarios. See each linked section for the configuration details.

1. Create a [local object store](#create-a-local-object-store-with-s3) with dedicated Ceph pools. This option is recommended if a single object store is required, and is the simplest to get started.
2. Create [one or more object stores with shared Ceph pools](#create-local-object-stores-with-shared-pools). This option is recommended when multiple object stores are required.
3. Connect to an [RGW service in an external Ceph cluster](#connect-to-an-external-object-store), rather than create a local object store.
4. Configure [RGW Multisite](#object-multisite) to synchronize buckets between object stores in different clusters.

!!! note
    Updating the configuration of an object store between these types is not supported.

> Rook has the ability to either deploy an object store in Kubernetes or to connect to an external RGW service.

Reviewer: There needs to be an empty line before the new paragraph, or this won't render online correctly.

Most commonly, the object store will be configured in Kubernetes by Rook.

> Alternatively, if you have an existing Ceph cluster with [Rados Gateways](https://docs.ceph.com/en/quincy/radosgw/index.html), see the [external section](#connect-to-an-external-object-store) to consume it from Rook.

Reviewer: ```suggestion
Alternatively see the [external section](#connect-to-an-external-object-store) to consume an existing Ceph cluster with [Rados Gateways](https://docs.ceph.com/en/quincy/radosgw/index.html) from Rook.
```

Multisite also allows object stores to be independent and isolated from other object stores in a cluster.

For more information on multisite please read the [ceph multisite overview](ceph-object-multisite.md) for how to run it.

## Using Swift and Keystone

> You can also access an object store using the [Swift API](https://developer.openstack.org/api-ref/object-store/index.html). 

Reviewer: "you"

Author: https://github.com/rook/rook/pull/13807#discussion_r1557848447

Multisite also allows object stores to be independent and isolated from other object stores in a cluster.

For more information on multisite please read the [ceph multisite overview](ceph-object-multisite.md) for how to run it.

## Using Swift and Keystone

> You can also access an object store using the [Swift API](https://developer.openstack.org/api-ref/object-store/index.html).

Reviewer: ```suggestion
It is possible to access an object store using the [Swift API](https://developer.openstack.org/api-ref/object-store/index.html).
```
